docs/hello_nf-core/03_add_module.md

```diff
@@ -1,63 +1,893 @@
+
+### 1.9. Wire up CAT_CAT in the workflow
+
+Now we need to modify our workflow code to use `CAT_CAT` instead of `collectGreetings`. Since `CAT_CAT` requires metadata tuples, we need to create a metadata map and combine it with our files.
+
+Open [core-hello/workflows/hello.nf](core-hello/workflows/hello.nf) and modify the workflow logic in the `main` block:
+
+=== "After"
+
+    ```groovy title="core-hello/workflows/hello.nf" linenums="26" hl_lines="7-17"
+        // emit a greeting
+        sayHello(ch_samplesheet)
+
+        // convert the greeting to uppercase
+        convertToUpper(sayHello.out)
+
+        // collect all the greetings into one file using nf-core cat/cat module
+        // create metadata map with batch name as the ID
+        def cat_meta = [ id: params.batch ]
+        ch_for_cat = convertToUpper.out.collect().map { files -> tuple(cat_meta, files) }
+
+        CAT_CAT(ch_for_cat)
+
+        // generate ASCII art of the greetings with cowpy
+        // Extract the file from the tuple since cowpy doesn't use metadata yet
+        ch_for_cowpy = CAT_CAT.out.file_out.map{ meta, file -> file }
+        cowpy(ch_for_cowpy, params.character)
+    ```
+
+=== "Before"
+
+    ```groovy title="core-hello/workflows/hello.nf" linenums="26" hl_lines="7-15"
+        // emit a greeting
+        sayHello(ch_samplesheet)
+
+        // convert the greeting to uppercase
+        convertToUpper(sayHello.out)
+
+        // collect all the greetings into one file
+        collectGreetings(convertToUpper.out.collect(), params.batch)
+
+        // emit a message about the size of the batch
+        collectGreetings.out.count.view { "There were $it greetings in this batch" }
+
+        // generate ASCII art of the greetings with cowpy
+        cowpy(collectGreetings.out.outfile, params.character)
+    ```
+
+Let's break down what we changed:
+
+1. **Created metadata**: `def meta = [ id: params.batch ]` creates a Groovy-style map with an `id` field set to our batch name
+2. **Created a tuple channel**: `ch_for_cat = convertToUpper.out.collect().map { files -> tuple(meta, files) }` combines the metadata and collected files into the tuple format expected by `CAT_CAT`
+3. **Called CAT_CAT**: Replaced `collectGreetings(...)` with `CAT_CAT(ch_for_cat)`
+4. **Extracted file for cowpy**: Since `cowpy` doesn't accept metadata tuples yet, we extract just the file from the tuple using `.map{ meta, file -> file }` and store it in `ch_for_cowpy`
+5. **Called cowpy**: Pass the extracted file channel to `cowpy` along with the character parameter
+6. **Removed count view**: The `cat/cat` module doesn't emit a count, so we removed that line
+
+!!! note
+
+    We removed the `collectGreetings.out.count.view { ... }` line because the nf-core `cat/cat` module doesn't provide a count of files. If you want to keep this functionality, you would need to count the files before calling `CAT_CAT`.
```

Comment:
Should we just remove this from the starter workflow entirely? It's something we use in Hello Nextflow to make a point but it doesn't really add anything except noise here. It would simplify things to not have it at all.